src/k8s/pkg/k8sd/api/certificates_status.go

```
@@ -0,0 +1,241 @@
+		if err != nil {
+			return response.InternalError(fmt.Errorf("failed to read datastore certificates: %w", err))
+		}
+		certificates = append(certificates, dataStoreCerts...)
+	}
+
+	updateExternallyManaged(authorities, certificates)
+	return response.SyncResponse(true, apiv1.CertificatesStatusResponse{
+		Certificates:           certificates,
+		CertificateAuthorities: authorities,
+	})
+}
+
+// getCertsStatusWorker collects certificate status information for worker
+// nodes. It reads worker certificates and kubeconfig certificates.
+func getCertsStatusWorker(s state.State, r *http.Request, snap snap.Snap) response.Response {
+	certsNames := []string{"kubelet"}
+	nodeCerts, err := loadCertificateStatusesFromDir(snap.KubernetesPKIDir(), certsNames)
+	if err != nil {
+		return response.InternalError(fmt.Errorf("failed to read node certificates: %w", err))
+	}
+
+	kubeConfigs := []string{"kubelet.conf", "proxy.conf"}
+	kubeConfigCerts, err := readKubeconfigCertificates(snap.KubernetesConfigDir(), kubeConfigs)
+	if err != nil {
+		return response.InternalError(fmt.Errorf("failed to read kubeconfig certificates: %w", err))
+	}
+
+	var certificates []apiv1.CertificateStatus
+	certificates = append(certificates, nodeCerts...)
+	certificates = append(certificates, kubeConfigCerts...)
+
+	// NOTE: Worker certificates are inherently externally managed.
+	for i := range certificates {
+		cert := &certificates[i]
+		cert.ExternallyManaged = true
+	}
+
+	return response.SyncResponse(true, apiv1.CertificatesStatusResponse{
+		Certificates:           certificates,
+		CertificateAuthorities: []apiv1.CertificateAuthorityStatus{},
+	})
+}
+
+// readKubeconfigCertificates reads the client certificates from kubeconfig
+// files located in the specified directory. It returns a slice of
+// CertificateStatus for each valid kubeconfig file.
+func readKubeconfigCertificates(kubeconfigDir string, configs []string) ([]apiv1.CertificateStatus, error) {
```

Why does this function silently ignore unknown/non-existing kubeconfigs?
Shouldn't the caller know what configs exist and expect an error if one does not?

---

You're right, I'll refactor the code. This is a leftover from my initial version, which I have already refactored a few times 😄